From 720fa339ce25bf818020d36b09bf28f7776837a0 Mon Sep 17 00:00:00 2001
From: Rick Farrington <rfarrington@marvell.com>
Date: Mon, 22 Jun 2020 13:45:15 -0400
Subject: [PATCH 580/767] mailbox: mvl_mhu: enable additional required
 configuration items

commit ac6680d7307384e333b57da712cf2323818d8f2e from
git@git.assembla.com:cavium/WindRiver.linux.git

Change-Id: I607ffc4d06da3b9d82cde92ac05b99a5fd52b441
Signed-off-by: Rick Farrington <rfarrington@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/30540
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
[Kevin: Fix the typo and indentation.]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/mailbox/Kconfig | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/mailbox/Kconfig b/drivers/mailbox/Kconfig
index efa7091caef3..aa3c299e4691 100644
--- a/drivers/mailbox/Kconfig
+++ b/drivers/mailbox/Kconfig
@@ -24,11 +24,20 @@ config IMX_MBOX
 
 config MVL_MHU
 	tristate "MVL MHU Mailbox"
+	select ARM_SCMI_PROTOCOL
+	select ARM_SCMI_CPUFREQ
+	select SENSORS_ARM_SCMI
+	select COMMON_CLK_SCMI
+	select ARM_SCMI_POWER_DOMAIN
+	default n
 	help
 	  Say Y here if you want to build the MVL MHU controller driver.
 	  The controller has 2 mailbox channels, the last of which can be
 	  used in secure mode only. Thi mail box uses internal CPC RAM for
 	  creating the mailbox memory.
+	  Setting the "Default CPUFreq governor (userspace)" is also required
+	  in order to change the CPU frequency via Linux userspace commands
+	  (see CPU Power Management/Frequency Scaling).
 
 config PLATFORM_MHU
 	tristate "Platform MHU Mailbox"
-- 
2.31.1

